src/engine/cameras/GameCamera.cpp

```diff
@@ -0,0 +1,92 @@
+#include <libultraship.h>
+#include <libultra/gbi.h>
+#include "GameCamera.h"
+#include "port/interpolation/FrameInterpolation.h"
+#include "engine/Matrix.h"
+#include "port/Game.h"
+
+extern "C" {
+#include "main.h"
+#include "code_800029B0.h"
+#include "camera.h"
+}
+
+size_t GameCamera::_count = 0;
+
+GameCamera::GameCamera() {
+    _camera = &cameras[_count];
+    _camera->cameraId = _count;
+    _camera->perspectiveMatrix = &PerspectiveMatrix;
+    _camera->lookAtMatrix = &LookAtMatrix;
+
+    _count += 1;
+}
+GameCamera::GameCamera(FVector pos, s16 rot, u32 mode) {
+    _camera = &cameras[_count];
+    _camera->cameraId = _count;
+    _camera->perspectiveMatrix = &PerspectiveMatrix;
+    _camera->lookAtMatrix = &LookAtMatrix;
+    _camera->renderMode = RENDER_TRACK_SECTIONS;
+    ProjMode = ProjectionMode::PERSPECTIVE;
+    bActive = true;
+    camera_init(Vec3f{pos.x, pos.y, pos.z}, rot, mode, _camera->cameraId);
+
+    _count += 1;
+}
+
+void GameCamera::Tick() {
+    if (!bActive) { return; }
+    if (nullptr == _camera) {
+        bActive = false;
+        return;
+    }
+
+    func_8001EE98(&gPlayers[_camera->playerId], _camera, _camera->cameraId);
+}
+
+void GameCamera::SetActive(bool state) {
+    bActive = state;
+}
+
+bool GameCamera::IsActive() {
+    return bActive;
+}
+
+Camera* GameCamera::Get() {
+    return _camera;
+}
+
+void GameCamera::SetProjectionMode(GameCamera::ProjectionMode mode) {
+    ProjMode = mode;
+}
+
+Mtx* GameCamera::GetPerspMatrix() {
+    return &PerspectiveMatrix;
+}
+
+Mtx* GameCamera::GetLookAtMatrix() {
+    return &LookAtMatrix;
+}
+
+void GameCamera::Setup() {
```

Comment:
Hmm may SetMatrix ?

Reply:
Lemme think about this one.
I suppose it's really `Draw()` I was always thinking like we were 'setting up the camera settings' but that's not really accurate. As this is what translates position and rotation of the camera.

SetMatrix does not seem descriptive enough to me. But I dunno. Could be good too.